### PR TITLE
chore(ci): Use reusable spellcheck workflow

### PR DESCRIPTION
**Pull request by phated**
# Related issue(s)

Resolves # 

# Description

This switches to a reusable spellcheck workflow (PRed at https://github.com/noir-lang/.github/pull/5) since we've added this to acvm too. 

I'll need to update the reference once the other PR is merged.

## Test additions / changes

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

**Comment by phated** [comment]
Tangent... do we want this to be categorized under the `Rust` workflow instead of the `Pull Request` workflow? It also lints markdown files 🤔

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -16,12 +16,5 @@ jobs:
...
-    steps:
-      - uses: actions/checkout@v3
-      - uses: streetsidesoftware/cspell-action@v2
-        with:
-          files: |
-            **/*.{md,rs}
-          incremental_files_only: true # Run this action on files which have changed in PR
-          strict: false # Do not fail, if a spelling mistake is found (This can be annoying for contributors)
+    name: spellcheck
+    uses: noir-lang/.github/.github/workflows/spellcheck.yml@phated/reusable-spellcheck
```
Comment (phated) [inline .github/workflows/rust.yml:20]:
```suggestion
    uses: noir-lang/.github/.github/workflows/spellcheck.yml@main
```
Once https://github.com/noir-lang/.github/pull/5 is merged

**Comment by kevaundray** [comment]
Yeah good question -- probably PR workflow since it will check markdown files

**Comment by phated** [comment]
> probably PR workflow since it will check markdown files

I just tried it and then I forgot that we run the `Pull Request` workflow on `pull_request_target` which would make it difficult to update the `cspell.json` file. Let's leave it in `rust.yml` for the time being.

**Comment by kevaundray** [comment]
Sounds good :y

**Review by kevaundray — APPROVED** [review]

**Pull request merged**